plugins/account_history/plugin.cpp

```diff
@@ -32,523 +32,400 @@ if( options.count(name) ) { \
-                ) {
+            auto itr = idx.lower_bound(boost::make_tuple(account, uint32_t(-1)));
+            uint32_t sequence = 0;
+            if (itr != idx.end() && itr->account == account) {
+                sequence = itr->sequence + 1;
+            }
 
-                obj.trx_id = _note.trx_id;
-                obj.block = _note.block;
-                obj.trx_in_block = _note.trx_in_block;
-                obj.op_in_trx = _note.op_in_trx;
-                obj.virtual_op = _note.virtual_op;
-                obj.timestamp = _db.head_block_time();
-                //fc::raw::pack( obj.serialized_op , _note.op);  //call to 'pack' is ambiguous
-                auto size = fc::raw::pack_size(_note.op);
-                obj.serialized_op.resize(size);
-                fc::datastream<char *> ds(obj.serialized_op.data(), size);
-                fc::raw::pack(ds, _note.op);
+            database.create<account_history_object>([&](account_history_object& history) {
+                history.account = account;
+                history.sequence = sequence;
+                history.op = op.id;
             });
         }
+    };
 
-        auto hist_itr = hist_idx.lower_bound(boost::make_tuple(item, uint32_t(-1)));
-        uint32_t sequence = 0;
-        if (hist_itr != hist_idx.end() && hist_itr->account == item) {
-                sequence = hist_itr->sequence + 1;
+    struct plugin::plugin_impl final {
+    public:
+        plugin_impl( )
+            : database_(appbase::app().get_plugin<chain::plugin>().db()) {
         }
 
-        _db.create<account_history_object>([&](account_history_object &ahist) {
-            ahist.account = item;
-            ahist.sequence = sequence;
-            ahist.op = new_obj->id;
-        });
-    }
-};
+        ~plugin_impl() = default;
 
-struct operation_visitor_filter : operation_visitor {
-    operation_visitor_filter(golos::chain::database &db, const golos::chain::operation_notification &note, const golos::plugins::account_history::operation_object *&n, std::string i, const flat_set<string> &filter, bool blacklist, uint32_t start_block)
-        : operation_visitor(db, note, n, i), _filter(filter), _blacklist(blacklist), _start_block(start_block) {
-    }
+        golos::chain::database & database() {
+            return database_;
+        }
 
-    const flat_set<string> &_filter;
-    bool _blacklist;
-    uint32_t _start_block;
-
-    template <typename T>
-    void operator()(const T &op) const {
-        if(_db.head_block_num() >= _start_block) {
-            if (_filter.find(fc::get_typename<T>::name()) != _filter.end()) {
-                if (!_blacklist)
-                    operation_visitor::operator()(op);
-            } else {
-                if (_blacklist)
-                    operation_visitor::operator()(op);
+        void on_operation(const golos::chain::operation_notification& note) {
+            if (!note.stored_in_db) {
+                return;
```

Comment:
Will `on_operation` be called again if we exit here ?

Reply:
No, it will be not called again.